src/shim.ts

```diff
@@ -6,43 +21,166 @@ import {
+     *     // orchestrator body
+     * });
+     * ```
+     */
+    export function orchestration(functionName: string, handler: OrchestrationHandler): void {
+        azFuncApp.generic(functionName, {
+            trigger: trigger.orchestration(),
+            handler: createOrchestrator(handler),
+        });
+    }
+
+    /**
+     * Registers a function as a Durable Entity for your Function App.
+     *
+     * @param functionName the name of your new durable entity
+     * @param handler the function that should act as an entity
+     *
+     * @example Register a counter entity
+     * ```javascript
+     * const df = require("durable-functions");
+     *
+     * df.entity('Counter', function (context) {
+     *     // entity body
+     * });
+     * ```
+     */
+    export function entity<T = unknown>(functionName: string, handler: EntityHandler<T>): void {
+        azFuncApp.generic(functionName, {
+            trigger: trigger.entity(),
+            handler: createEntityFunction(handler),
+        });
+    }
+
+    /**
+     * Registers a function as an Activity Function for your Function App
+     *
+     * @param functionName the name of your new activity function
+     * @param options the configuration options for this activity,
+     * specifying the handler and the inputs and outputs
+     *
+     * @example Register an activity function
+     * ```javascript
+     * const df = require("durable-functions");
+     *
+     * df.activity('MyActivity', {
+     *   handler: function (context) {
+     *      // activity body
+     *   }
+     * });
+     * ```
+     */
+    export function activity(functionName: string, options: ActivityOptions): void {
+        azFuncApp.generic(functionName, {
+            trigger: trigger.activity(),
+            ...options,
+        });
+    }
+}
+
+/**
+ * The root namespace to help create durable trigger configurations
+ */
+export namespace trigger {
+    /**
+     * @returns a durable activity trigger
+     */
+    export function activity(): ActivityTrigger {
+        return azFuncTrigger.generic({
+            type: "activityTrigger",
+        }) as ActivityTrigger;
+    }
+
+    /**
+     * @returns a durable orchestration trigger
+     */
+    export function orchestration(): OrchestrationTrigger {
+        return azFuncTrigger.generic({
+            type: "orchestrationTrigger",
+        }) as OrchestrationTrigger;
+    }
+
+    /**
+     * @returns a durable entity trigger
+     */
+    export function entity(): EntityTrigger {
+        return azFuncTrigger.generic({
+            type: "entityTrigger",
+        }) as EntityTrigger;
+    }
+}
+
+/**
+ * The root namespace to help create durable input configurations
+ */
+export namespace input {
+    /**
+     * @returns a durable client input configuration object
+     */
+    export function durableClient(): DurableClientInput {
+        return azFuncInput.generic({
+            type: "orchestrationClient",

```

Comment:
I think the preferred `type` here should be `"durableClient"`.
I think `orchestrationClient` is a bit of a legacy name.,

Reply:
I actually don't think this is possible (or at least it would require more changes at the extensions level). This string translates to what you would put in a `function.json` file. Since the name of the binding is `"orchestrationClient"`, this is what I have here. Otherwise, it wouldn't register this binding. Although I'm not sure where the names of bindings come from to begin with? 🤔

Reply:
My bad I didn't realize both were already valid names for this binding! We should update our templates to reflect the new preferred name because I've only ever seen `orchestationClient` (at least for node templates)

Reply:
I can confirm that there's a "durableClient" binding that is equivalent to the "orchestrationClient" :). See here: https://learn.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-bindings?tabs=python%2C2x-durable-functions#client-usage

Reply:
Updated this to `durableClient` though